### PR TITLE
Add option to allow table height to grow without specified container height

### PR DESCRIPTION
**Pull request by LydiaBarnes**
Currently table content height is calculated to never go higher than its container height so that it can scroll within it. But sometimes a table that just grows infinitely without having to specify a height on the container is desired.
This change adds a flag passed in through the component to tell it whether to restrict to the container height or not.

Table without container height set and flag = true:
screen shot 2016-02-18 at 7 00 13 pm

Table without container height set and flag = false:
screen shot 2016-02-18 at 6 58 34 pm

@Addepar/pants @Addepar/fire

### CODE REVIEW

**Comment by LydiaBarnes** [comment]
Major credit to @thangdinh  for helping me with this

File: addon/components/ember-table.js [hunk 1]
```diff
@@ -75,6 +75,8 @@ StyleBindingsMixin, ResizeHandlerMixin, {
   // through ctrl/cmd-click or shift-click).
   selectionMode: 'single',
 
+  restrictToContainerHeight: true,
```
Comment (embooglement) [inline addon/components/ember-table.js]: I would name this something more like `isRestrictedToContainerHeight`

File: addon/components/ember-table.js [hunk 2]
```diff
@@ -409,7 +411,13 @@ StyleBindingsMixin, ResizeHandlerMixin, {
...
         this.get('_headerHeight') + this.get('_footerHeight');
-    return Math.min(contentHeight, height);
+
+    if (this.get('restrictToContainerHeight')) {
+      return Math.min(contentHeight, height);
+    }
+    else {
+      return contentHeight;
+    }
   }).property('_height', '_tableContentHeight', '_headerHeight',
```
Comment (embooglement) [inline addon/components/ember-table.js]: You are missing the new property in the dependent keys.

Comment (LydiaBarnes) [inline addon/components/ember-table.js]: done!

File: dist/ember-table.css [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /*!
 * ember-table v0.5.1
-* Copyright 2012-2015 Addepar Inc.
+* Copyright 2012-2016 Addepar Inc.
```
Comment (hanakimis) [inline dist/ember-table.css]: :+1:

**Comment by cyril-sf** [comment]
@LydiaBarnes Travis.ci is reporting that tests are failing.

**Comment by cyril-sf** [comment]
```not ok 1 PhantomJS 1.9 - global failure
    ---
        actual: >
            null
        message: >
            Error: Could not find module ember/resolver```

Actually, it doesn't seem to be related to your PR

**Comment by LydiaBarnes** [comment]
retest

**Comment by thangdinh** [comment]
Is it because this PR is based off Addepar branch?

File: dist/ember-table.js [hunk 4]
```diff
@@ -16,156 +16,72 @@ var define, requireModule, require, requirejs;
   } else {
     _isArray = Array.isArray;
   }
-
-  var registry = {}, seen = {};
+  
+  var registry = {}, seen = {}, state = {};
   var FAILED = false;
 
-  var uuid = 0;
```
Comment (cyril-sf) [inline dist/ember-table.js]: @LydiaBarnes none of that should have changed with your PR. You need to rebuild the dist files.

**Pull request closed without merging**